lib/internal/test_runner/test.js

```diff
@@ -754,6 +762,12 @@ class Test extends AsyncResource {
 
       if (coverage) {
         reporter.coverage(nesting, loc, coverage);
+        const failedCoverage = ArrayPrototypeFilter(
+          ObjectKeys(coverage.threshold), (key) => !coverage.threshold[key]);
+        if (failedCoverage.length > 0) {
+          console.error(`test coverage failed for ${ArrayPrototypeJoin(failedCoverage, ', ')}`);
+          process.exit(1);
```

Comment:
here I added the logic to exit the test when coverage threshold doesn't match.

running test will be exited with status code 1.

output:
```js
TAP version 13
# Subtest: hello test
ok 1 - hello test
  ---
  duration_ms: 3.603875
  ...
test coverage failed for lines, functions, branches
```

cc @ljharb @meyfa